### DISCUSSION TITLE
Iterate through an xml

### DISCUSSION THREAD

**Discussion by patrickdorival** [downforcetech/fluidxml.php#22] [Questions]
I'm finding it difficult to iterate through an xml that has an unknown contents. 
Accessing it via nodeName => nodeValue is not producing the correct results.

Where in the documentation does it state how to do this, for example in a foreach loop over a queried element that has many child elements that I need to retrieve the name and value as well as any attributes?

**Comment by daniele-orlando** [comment]
One possible solution of the many allowed by XPath.

```
$doc->query('/doc/books/*')
    ->each(function ($i, $domnode) {});
```

Learning more about XPath will help you in this.

**Comment by patrickdorival** [comment]
Hmm I don't think the issue is knowing more about XPath but okay. I've switched back to SimpleXML and its sorted.
